### PR TITLE
loosen azure sdk dependencies in azurearm cloud driver

### PR DESCRIPTION
**Pull request by jgleissner**
### What does this PR do?
This PR removes the dependency on azure-cli from the azurearm cloud driver, which is not used at all. It also changes the driver to import the azure-storage sdk package in lieu of the multiapi storage package, in case the latter is not available.

### What issues does this PR fix or reference?
Allows the azurearm cloud driver to be used on platforms that do not have the azure-cli or azure-multiapi-storage packages.

### Tests written?
Manually tested with azure-storage 0.36 and azure-multiapi-storage 0.1.4.

### Commits signed with GPG?
Yes

### CODE REVIEW

**Comment by rallytime** [comment]
Welcome @jgleissner! Thank you for contributing.

Can you fix the lint error? https://jenkinsci.saltstack.com/job/pr-lint/job/PR-49696/1/warnings52Result/

File: salt/cloud/clouds/azurearm.py [hunk 1]
```diff
@@ -117,9 +118,12 @@
     from azure.mgmt.storage import StorageManagementClient
     from azure.mgmt.web import WebSiteManagementClient
     from msrestazure.azure_exceptions import CloudError
-    from azure.multiapi.storage.v2016_05_31 import CloudStorageAccount
-    from azure.cli import core
-    HAS_LIBS = LooseVersion(core.__version__) >= LooseVersion("2.0.12")
+    if pkgutil.find_loader('azure.multiapi'):
```
Comment (rallytime) [inline salt/cloud/clouds/azurearm.py]: What about using the `salt.utils.path.which` function instead here?

**Review by rallytime — COMMENTED** [review]
I have one small suggestion.

File: salt/cloud/clouds/azurearm.py (hunk 1 above)
Comment (jgleissner) [reply salt/cloud/clouds/azurearm.py]: Hi @rallytime , if I'm not mistaken, `salt.utils.path.which` looks for an executable, whereas this code looks for an importable module. I don't see how `which` would be useful here, or am I missing something?

**Review by rallytime — APPROVED** [review]

**Comment by rallytime** [comment]
Thank you @jgleissner!

**Pull request merged**